### DISCUSSION TITLE
question about snapshot

### DISCUSSION THREAD

**Discussion by calvin2021y** [rabbitmq/ra#300] [Q&A]
I am new to erlang and this project look great.

I have one question about snapshot:  how to provide myown snapshot/restore callback function like https://github.com/canonical/raft ?

 snapshot_callback should take state as argument, and return  snapshot.

restore_callback should take snapshot as argument, and return state.

**Comment by kjnilsson** [comment]
It’s not documented but there is a ra_snapshot behaviour you can implement
and configure.

On Thu, 7 Jul 2022 at 05:17, calvin2021y ***@***.***> wrote:

> I am new to erlang and this project look great.
>
> I have one question about snapshot: how to provide myown snapshot/restore
> callback function like https://github.com/canonical/raft ?
>
> snapshot_callback should take state as argument, and return snapshot.
>
> restore_callback should take snapshot as argument, and return state.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/ra/discussions/300>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFBPZ4NON2WAGTO7ENLVSZK6XANCNFSM5235MY4Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
*Karl Nilsson*
